### PR TITLE
[PKG-5526] Update to 4.1.0

### PR DESCRIPTION
**Pull request by ViridianMelody**
mmh3 4.1.0

**Destination channel:** defaults

### Links

- [PKG-5526]
- [Upstream repository](https://github.com/hajimes/mmh3)
- [Upstream changelog/diff](https://github.com/hajimes/mmh3/blob/master/CHANGELOG.md)

### Explanation of changes:

- Added upstream tests

[PKG-5526]: https://anaconda.atlassian.net/browse/PKG-5526?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,19 +27,27 @@ requirements:
     - python
 
 test:
+  source_files:
+    - tests
   imports:
     - mmh3
   requires:
     - pip
+    - mypy
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
Mypy is not needed

File: recipe/meta.yaml [hunk 2]
```diff
@@ -27,19 +27,26 @@ requirements:
...
     - mmh3
   requires:
     - pip
+    - pytest
   commands:
     - pip check
+    - pytest tests
 
 about:
   home: https://github.com/hajimes/mmh3
```
Comment (ifitchet) [inline recipe/meta.yaml:42]: I think the homepage has changed, check `pyproject.toml`

```suggestion
  home: https://pypi.org/project/mmh3/
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -27,19 +27,26 @@ requirements:
     - python
 
 test:
+  source_files:
+    - tests
   imports:
     - mmh3
   requires:
     - pip
+    - pytest
```
Comment (ifitchet) [inline recipe/meta.yaml:36]: Does it not use `mypy` in the tests?  Just curious as it is listed as a test dependency.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -27,19 +27,26 @@ requirements:
...
   commands:
     - pip check
+    - pytest tests
 
 about:
   home: https://github.com/hajimes/mmh3
   license: CC0-1.0
```
Comment (ifitchet) [inline recipe/meta.yaml]: The `LICENSE` looks like MIT...

Comment (ViridianMelody) [reply recipe/meta.yaml]: Good catch, thanks!

File: recipe/meta.yaml (hunk 3 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:36]: It doesn't. They run mypy in the [github build action](https://github.com/hajimes/mmh3/blob/ba23958677b4ba5153fd7d07f3f1d56cbdf56deb/.github/workflows/build.yml#L44-L45), but it's not in the tests.

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:42]: Umm, technically, the real home is https://github.com/hajimes/mmh3.

**Review by Arishamays1 — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (ifitchet) [reply recipe/meta.yaml:42]: @JeanChristopheMorinPerso how do we decide on the homepage?  The `pyproject.toml` points to PyPI (where the Homepage points to itself), the Github page also uses its documentation link (top right) to point to PyPI.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:42]: My logic is usually: website > source (github, etc) > PyPI as a last resort.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:42]: For reference, here are the top home_url (by domain name) in all the packages we maintain (not feedstock, packages) that have more than 50 hits:
```
                  home_url                  | total 
--------------------------------------------+-------
 github.com                                 |  5850
 CRAN.R-project.org                         |  4484
 x.org                                      |   186
 gnu.org                                    |   146
 docs.ropensci.org                          |   126
 developer.nvidia.com                       |   115
 r-project.org                              |   113
 bitbucket.org                              |    81
 gcc.gnu.org                                |    79
 airflow.apache.org                         |    78
 r-forge.r-project.org                      |    68
 gitlab.com                                 |    64
 pypi.org                                   |    64
 cran.r-project.org                         |    62
 llvm.org                                   |    61
 dirk.eddelbuettel.com                      |    55
```

Comment (ryanskeith) [reply recipe/meta.yaml:42]: This seems like an excellent policy question that we should note and discuss in our upcoming team meetings.

**Pull request merged**